src/routes/stops/[stopID]/+page.server.js

```diff
@@ -1,11 +1,11 @@
-import stopAPI from '$lib/RestAPI/stop.js';
-import arrivalDepartureAPI from '$lib/RestAPI/arrivalsAndDeparturesForStop.js';
+import arrivalsAndDeparturesForStop from '../../../lib/RestAPI/arrivalsAndDeparturesForStop.js';
+import stop from '../../../lib/RestAPI/stop.js';
 
 export async function load({ params }) {
```

Comment:
I think this function will be a lot clearer if we use your obaSdk abstraction directly, instead of the layer on top of it.

Reply:
I was considering deleting the RestApi folder and using the SDK directly in the api/routes folder.

Actually, I initially implemented it this way but reverted back because I noticed that src/routes/stops uses some functions from RestApi. If we want to use the SDK directly without any level of abstraction, we will need to check response.code !== 200 and handle errors in every file. So, I think the abstraction is beneficial for handling errors and other repetitive tasks. What do you think?

To sum up the current options:

1 - Delete the RestApi folder and use the SDK directly in src/routes, adding error handling in every fetch throughout the app.
2 - Keep the RestApi and continue with this pattern of abstraction.

Reply:
So, we can use abstractions as level of error handling and make it reusable

Reply:
That's a fair point. I spent some time thinking about this, and I think that a good middle ground would be to add a new function to obaSdk.js called `handleOBAResponse()` (or something similar—I'm open to suggestions on naming 😄) that takes care of the repeated handling of 200 and non-200 status codes that we see throughout the API layer:

`return handleOBAResponse(response, 'Unable to fetch shape.');`


```
function handleOBAResponse(response, errorMessage) {
	if (response.code !== 200) {
		return error(500, errorMessage);
	}

	return json(response);
}
```

Reply:
I think this is a great solution. I was thinking that we could add a pattern and use it across all API fetching.

Here's what I will do:

1. Complex APIs will have their logic separated into the RestApi folder.
2: For simple API fetching, like shape, I will use ObaSdk directly in the routes/api folder.
3. I will add an obaHandleApi function and use it for any API calls in the app

Reply:
I have added the new modifications and removed the RestApi folder.

@aaronbrethorst